academic_observatory_workflows/workflows/oa_web_workflow.py

```diff
@@ -403,6 +410,111 @@ def to_dict(self) -> Dict:
+        clearbit_download_logo(company_url=url, file_path=file_path, size=width, fmt=fmt)
+    if os.path.isfile(file_path):
+        logo_path = make_logo_url(category="institution", entity_id=ror_id, size=size, fmt=fmt)
+
+    return ror_id, logo_path
+
+
+def get_wiki_descriptions(titles: Dict[str, str]) -> List[Tuple[str, str]]:
+    """Get the wikipedia descriptions for the given titles.
+
+    :param titles: Dict with titles as keys and id's (either ror_id or alpha3 country code) as values
+    :return: List with tuples (id, wiki description)
+    """
+    titles_arg = []
+    for title, entity_id in titles.items():
+        # URL encode title if it is not encoded yet
+        if title == urllib.parse.unquote(title):
+            titles_arg.append(urllib.parse.quote(title))
+        # Append title directly if it is already encoded and not empty
+        else:
+            titles_arg.append(title)
+
+    # Confirm that there is a max of 20 titles, the limit for the wikipedia API
+    assert len(titles_arg) <= 20
+
+    # Extract descriptions using the Wikipedia API
+    url = f"https://en.wikipedia.org/w/api.php?action=query&format=json&prop=extracts&titles={'%7C'.join(titles_arg)}&redirects=1&exintro=1&explaintext=1"
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise AirflowException(f"Unsuccessful retrieving wikipedia extracts, url: {url}")
+    response_json = response.json()
+    pages = response_json["query"]["pages"]
+
+    # Create mapping between redirected page title and original page title
+    redirects = {}
+    for title in response_json["query"].get("redirects", []):
+        redirects[title["to"]] = title["from"]
+
+    # Create mapping between entity_id and decoded, alpha-only page title.
+    alpha_titles = {re.sub("[^A-Za-z]+", "", urllib.parse.unquote(k)): v for k, v in titles.items()}
+    descriptions = []
+    for page_id, page in pages.items():
+        page_title = page["title"]
+        # Get page_title from redirect if it is present
+        page_title = redirects.get(page_title, page_title)
+
+        # Match entity_id and page_title
+        entity_id = alpha_titles[re.sub("[^A-Za-z]+", "", page_title)]
+
+        # Get description and clean up
+        description = page.get("extract", "")
+        if description:
+            description = remove_text_between_brackets(description)
+            description = shorten_text_full_sentences(description)
+
+        descriptions.append((entity_id, description))
+    return descriptions
+
+
+def remove_text_between_brackets(text: str) -> str:
+    new_text = ""
+    nested = 0
+    for char in text:
+        if char == "(":
+            nested += 1
+            new_text = new_text.rstrip(" ")
+        elif (char == ")") and nested:
+            nested -= 1
+        elif nested == 0:
+            new_text += char
+    return new_text
+
+
+def shorten_text_full_sentences(text: str, *, char_limit: int = 500) -> str:
```

Comment:
```suggestion
def shorten_text_full_sentences(text: str, *, char_limit: int = 300) -> str:
```

Just to get slightly shorter descriptions in the website.